internal/storage/persistence/persistence.go

```diff
@@ -27,4 +34,12 @@ type Storage interface {
 	ListLocations(ctx context.Context) (goastorage.LocationCollection, error)
 	ReadLocation(ctx context.Context, locationID uuid.UUID) (*goastorage.Location, error)
 	LocationAIPs(ctx context.Context, locationID uuid.UUID) (goastorage.AIPCollection, error)
+
+	// Workflow.
+	CreateWorkflow(context.Context, *types.Workflow) error
```

Comment:
I'm not sure I have much to add to the "DB IDs vs UUIDs" discussion, but here's a suggestion. Would it make sense to update this method to look up the AIP's database ID directly from its UUID? That would let you do the insert in a more atomic way (using a transaction or an `INSERT INTO ... SELECT ...` if supported by entgo), avoiding potential race conditions where the AIP record might change or be deleted between the lookup and the insert - perhaps that kind of concurrency is not a concern here? Beyond ensuring data consistency, that would also save you from having to implement `AIPDBID` and the extra roundtrip with the database.

Reply:
Thanks, not sure why I didn't do it that way, seems so obvious now that you say it ;)